src/templates/Program/Program.jsx

```diff
@@ -140,118 +124,70 @@ const Program = ({
     return modifiedPlannings;
   }, [plannings]);
 
-  // Initialize filters once we got plannings from Swapcard
-  useEffect(() => {
-    // const places = [];
-    const categories = [];
-    const eventTypes = [];
-
-    const addChoices = (value, array) => {
-      if (value === null || array.some((v) => v === value)) return;
-
-      array.push(value);
-    };
-
-    program.forEach((session) => {
-      // Get all places for filters
-      // addChoices(session.place, places);
-      // Get all categories for filters
-      session.categories.forEach((category) => {
-        addChoices(category, categories);
-      });
-      // Get all types for filters
-      addChoices(session.type, eventTypes);
-    });
-
-    if (filters.length > 0) {
-      filterDispatcher({
-        type: 'UPDATE',
-        options: {
-          name: 'categories',
-          values: categories.map((value) => ({
-            name: categoriesMap[value],
-            value,
-          })),
-        },
-      });
-
-      filterDispatcher({
-        type: 'UPDATE',
-        options: {
-          name: 'type',
-          values: eventTypes.map((value) => ({
-            name: eventTypesMap[value],
-            value,
-          })),
-        },
-      });
-
-      return;
+  const groupedByTimeProgram = {};
+  program.forEach((session) => {
```

Comment:
Tu pourrais utiliser un reduce ici pour que ça soit un peu plus clean et de mettre en mémoire le résultat en utilisant un `useMemo`.
```suggestion
  const groupedByTimeProgram = useMemo(
    () =>
      program.reduce((acc, current) => {
        ...
      }, {}),
    [program]
  );
```